rust/src/builders/metadata_builder.rs

```diff
@@ -0,0 +1,46 @@
+use crate::*;
+
+#[wasm_bindgen]
+pub struct TxMetadataBuilderResult {
+    pub(crate) metadata: GeneralTransactionMetadata,
+}
+
+#[wasm_bindgen]
+pub struct TxMetadataBuilder {
+    metadata: GeneralTransactionMetadata,
+}
+
+#[wasm_bindgen]
+impl TxMetadataBuilder {
+    pub fn new() -> Self {
```

Comment:
Probably it's better to have two constructors:
1. `pub fn empty()` that works like what you have now
2. `pub fn from(metadata: &GeneralTransactionMetadata)`

Reply:
That's a great suggestion. Let me add that